python/GafferSceneUI/ShaderTweaksUI.py

```diff
@@ -349,21 +350,22 @@ def __addTweak( self, name, plugTypeOrValue ) :
 def __setShaderFromAffectedMenuDefinition( menu ) :
 
 	plugValueWidget = menu.ancestor( GafferUI.PlugValueWidget )
-	return __setShaderFromPathsMenuDefinition( plugValueWidget, _pathsFromAffected( plugValueWidget ) )
+	useFullAttr = plugValueWidget.getPlug().node()["localise"].getValue()
+	return __setShaderFromPathsMenuDefinition( plugValueWidget, _pathsFromAffected( plugValueWidget ), useFullAttr )
 
 def __setShaderFromSelectionMenuDefinition( menu ) :
 
 	plugValueWidget = menu.ancestor( GafferUI.PlugValueWidget )
-	return __setShaderFromPathsMenuDefinition( plugValueWidget, _pathsFromSelection( plugValueWidget ) )
+	return __setShaderFromPathsMenuDefinition( plugValueWidget, _pathsFromSelection( plugValueWidget ), False )
```

Comment:
I don't understand why it's preferable to pass `useFullAttr = False` here. If we didn't, we could strip out all this extra complexity and just deal with `localise` in a single place in `_shaderAttributes()`, where we're already doing the correct context handling.